runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java

```diff
@@ -142,6 +145,33 @@ public void checkForValidStatus()
 		}
 	}
 
+	private void checkForAreaUpdate()
+	{
+		if (client.getLocalPlayer().getWorldLocation().getRegionID() == 0)
+		{
+			return;
+		}
+
+		final DiscordGameEventType discordGameEventType = DiscordGameEventType.fromLocation(client.getLocalPlayer().getWorldLocation().getRegionID());
+
+		if (discordGameEventType == null || lastEvent != null && lastEvent.name() == discordGameEventType.name())
+		{
+			return;
+		}
+
+		switch (discordGameEventType.getDiscordAreaType())
+		{
+			case BOSSES: if (!config.showBossActivity()) return;
```

Comment:
The cases will leak across if the config is not on for that area type.

You could do something like..
```
boolean enabled = true;
switch(discordGameEventType.getDiscordAreaType()){
case BOSSES: enabled = config.showBossActivity(); break;
case CITIES: enabled = config.showCityActivity(); break;
...
}
if (!enabled) {
return;
}
```